test/unit/test_anal_meta.c

```diff
@@ -0,0 +1,615 @@
+	r_meta_set_string (anal, R_META_TYPE_COMMENT, 0x100, "vera gemini");
+	r_meta_set_with_subtype (anal, R_META_TYPE_STRING, R_STRING_ENC_UTF8, 0x200, 0x30, "true confessions");
+
+	RPVector *items = r_meta_get_all_at (anal, 0x100);
+	mu_assert_eq (r_pvector_len (items), 2, "all count");
+	void **it;
+	bool found[2] = { 0 };
+	r_pvector_foreach (items, it) {
+		RAnalMetaItem *item = ((RIntervalNode *)*it)->data;
+		switch (item->type) {
+		case R_META_TYPE_DATA:
+			found[0] = true;
+			break;
+		case R_META_TYPE_COMMENT:
+			found[1] = true;
+			break;
+		default:
+			break;
+		}
+	}
+	mu_assert ("meta 0", found[0]);
+	mu_assert ("meta 1", found[1]);
+	r_pvector_free (items);
+
+	items = r_meta_get_all_at (anal, 0xff);
+	mu_assert_eq (r_pvector_len (items), 0, "all count");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_at (anal, 0x101);
+	mu_assert_eq (r_pvector_len (items), 0, "all count");
+	r_pvector_free (items);
+
+	r_anal_free (anal);
+	mu_end;
+}
+
+bool test_meta_get_all_in() {
+	RAnal *anal = r_anal_new ();
+
+	r_meta_set (anal, R_META_TYPE_DATA, 0x100, 4, NULL);
+	r_meta_set_string (anal, R_META_TYPE_COMMENT, 0x100, "vera gemini");
+	r_meta_set_with_subtype (anal, R_META_TYPE_STRING, R_STRING_ENC_UTF8, 0x200, 0x30, "true confessions");
+
+	RPVector *items = r_meta_get_all_in (anal, 0x100, R_META_TYPE_ANY);
+	mu_assert_eq (r_pvector_len (items), 2, "all count");
+	void **it;
+	bool found[2] = { 0 };
+	r_pvector_foreach (items, it) {
+		RAnalMetaItem *item = ((RIntervalNode *)*it)->data;
+		switch (item->type) {
+		case R_META_TYPE_DATA:
+			found[0] = true;
+			break;
+		case R_META_TYPE_COMMENT:
+			found[1] = true;
+			break;
+		default:
+			break;
+		}
+	}
+	mu_assert ("meta 0", found[0]);
+	mu_assert ("meta 1", found[1]);
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0x100, R_META_TYPE_COMMENT);
+	mu_assert_eq (r_pvector_len (items), 1, "all count");
+	RAnalMetaItem *item = ((RIntervalNode *)r_pvector_at (items, 0))->data;
+	mu_assert_streq (item->str, "vera gemini", "contents");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0x100, R_META_TYPE_DATA);
+	mu_assert_eq (r_pvector_len (items), 1, "all count");
+	item = ((RIntervalNode *)r_pvector_at (items, 0))->data;
+	mu_assert_eq (item->type, R_META_TYPE_DATA, "contents");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0x103, R_META_TYPE_DATA);
+	mu_assert_eq (r_pvector_len (items), 1, "all count");
+	item = ((RIntervalNode *)r_pvector_at (items, 0))->data;
+	mu_assert_eq (item->type, R_META_TYPE_DATA, "contents");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0xff, R_META_TYPE_ANY);
+	mu_assert_eq (r_pvector_len (items), 0, "all count");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0x101, R_META_TYPE_COMMENT);
+	mu_assert_eq (r_pvector_len (items), 0, "all count");
+	r_pvector_free (items);
+
+	items = r_meta_get_all_in (anal, 0x103, R_META_TYPE_DATA);
```

Comment:
haven't you already done this test at https://github.com/radareorg/radare2/pull/16716/commits/0c3cddc779dc59219e87b20b0d75bc4a78055bc7#diff-2bdaf63af99d8302c5148cd3897de99cR293 ?

Reply:
yep